### PR TITLE
New function : pgr_makeMaximalPlanar

### LINKED ISSUE #3141: GSOC 2026: New function pgr_makeMaximalPlanar to be added in pgrouting
## pgr_makeMaximalPlanar():

**makeMaximalPlanar()**: Maximal planar augmentation (triangulation) is an algorithm that takes a biconnected planar graph and adds edges until the graph becomes maximally planar. A graph is maximally planar if no more edges can be added without violating planarity, which results in a triangulated graph where every face is a triangle (satisfying Euler's formula E = 3V - 6). Triangulated planar graphs are required for advanced graph layout algorithms like Chrobak-Payne straight line drawing. This implementation uses the Boost Graph Library's `make_maximal_planar` algorithm with a time complexity of O(V + E), where V is the number of vertices and E is the number of edges. This will enhance pgRouting's capabilities in planar graph optimization and spatial rendering problems.

#### The algorithm:

- Works on **undirected** graphs.
- Requires the input graph to already be biconnected and planar.
- Adds new edges between vertices strictly preserving the planarity of the embedding.
- Outputs the set of edges added to the graph to triangulate it.
- Running time: O(V + E) where V is the number of vertices and E is the number of edges.

### Signature:

- pgr_makeMaximalPlanar()

```sql
pgr_makeMaximalPlanar(Edges SQL)

Returns set of (seq, start_vid, end_vid)
OR EMPTY SET
```

## Parameters

| Parameter | Type | Description |
| --- | --- | --- |
| **Edges SQL** | `TEXT` | Inner SQL query, as described below. |

### Inner Query

**Edges SQL**: An SQL query returning a set of rows with the following columns:

| Column | Type | Default | Description |
| --- | --- | --- | --- |
| **id** | `ANY-INTEGER` | | Identifier of the edge. |
| **source** | `ANY-INTEGER` | | Identifier of the first endpoint vertex of the edge. |
| **target** | `ANY-INTEGER` | | Identifier of the second endpoint vertex of the edge. |
| **cost** | `ANY-NUMERICAL` | | Weight of the edge `(source, target)`. When negative, the edge does not exist. |
| **reverse_cost** | `ANY-NUMERICAL` | `-1` | Weight of the edge `(target, source)`. When negative, the edge does not exist. |

Where:

- `ANY-INTEGER` = `SMALLINT`, `INTEGER`, `BIGINT`
- `ANY-NUMERICAL` = `SMALLINT`, `INTEGER`, `BIGINT`, `REAL`, `FLOAT`

### Result Columns

Returns `SETOF (seq, start_vid, end_vid)`.

| Column | Type | Description |
| --- | --- | --- |
| **seq** | `BIGINT` | Sequential value starting from 1. |
| **start_vid** | `BIGINT` | Identifier of the first endpoint vertex of the added edge. |
| **end_vid** | `BIGINT` | Identifier of the second endpoint vertex of the added edge. |

### PR DESCRIPTION
**Pull request by Mohit242-bit**
This PR adds one new function named pgr_makeMaximalPlanar to pgrouting:

- `pgr_makeMaximalPlanar`: Returns the set of edges needed to make each connected component of a planar graph maximal planar (triangulated).

This is a final merge PR containing commits from several different PRs combined.

The intermediate PRs made on the [GSoC-pgRouting](https://github.com/pgRouting/GSoC-pgRouting) repository are as follows:

- pgRouting/GSoC-pgRouting#554
- pgRouting/GSoC-pgRouting#560
- pgRouting/GSoC-pgRouting#571
- pgRouting/GSoC-pgRouting#573
- pgRouting/GSoC-pgRouting#583
- pgRouting/GSoC-pgRouting#587
- pgRouting/GSoC-pgRouting#592
- pgRouting/GSoC-pgRouting#597
- pgRouting/GSoC-pgRouting#605
- pgRouting/GSoC-pgRouting#613
- pgRouting/GSoC-pgRouting#623
- pgRouting/GSoC-pgRouting#626
- pgRouting/GSoC-pgRouting#633

@pgRouting/admins

## New Features
- Added the experimental `pgr_makeMaximalPlanar` function for triangulating eligible connected components of undirected planar graphs.
- Added Planar family documentation, examples, diagrams, release notes, and usage details.
- Non-planar components and graphs with fewer than three vertices return no added edges.

## Tests
- Added coverage for edge cases, output types, inner queries, and regression scenarios.

### CODE REVIEW

**Review by cvvergara — APPROVED** [review]
Thanks for your work.
This approval is about work review.
To be merged it needs approval from PSC with a motion on Discourse. 
Any of the other members listed for review, once approved will do,

**Review by krashish8 — APPROVED** [review]
LGTM. Thanks for adding this! 🚀

**Review by sanak — APPROVED** [review]
LGTM!

File: pgtap/planar/makeMaximalPlanar/no_crash_test.pg [hunk 1]
```diff
@@ -0,0 +1,44 @@
+/* :file: This file is part of the pgRouting project.
+:copyright: Copyright (c) 2018-2026 pgRouting developers
+:license: Creative Commons Attribution-Share Alike 3.0 https://creativecommons.org/licenses/by-sa/3.0 */
+
+BEGIN;
+
+UPDATE edges SET cost = sign(cost), reverse_cost = sign(reverse_cost);
```
Comment (iosefa) [inline pgtap/planar/makeMaximalPlanar/no_crash_test.pg:7]: This sets all positive costs (both reverse and non-reverse) to 1.

**Review by iosefa — COMMENTED** [review]
I approve adding the function; thank you! But, I ran a test with unequal directional costs and I receive a segmentation fault. I noticed in the tests: https://github.com/pgRouting/pgrouting/pull/3134/changes#r4077365377  - which sets all costs, reverse and non-reverse, to 1. but removing that and creating a simple test where reverse cost ≠ cost breaks things. e.g. cost = 1, reverse_cost = 2. I just wanted to flag that. It might be worth fixing before merging this in.

File: pgtap/planar/makeMaximalPlanar/no_crash_test.pg (hunk 1 above)
Comment (cvvergara) [reply pgtap/planar/makeMaximalPlanar/no_crash_test.pg:7]: Actually it keeps the sign (1 or -1) does not take into account the lenght, that is done on all tests, to make sure the data is fixed after some other test might have modified it
```
select * from edges where cost = -1 or reverse_cost = -1;
 id | source | target | cost | reverse_cost | capacity | reverse_capacity | x1 | y1 | x2 | y2 
----+--------+--------+------+--------------+----------+------------------+----+----+----+----
  5 |     10 |     11 |    1 |           -1 |      130 |               -1 |  3 |  1 |  3 |  2
 12 |      8 |     12 |    1 |           -1 |      100 |               -1 |  2 |  3 |  3 |  3
 11 |     11 |     12 |    1 |           -1 |      130 |               -1 |  3 |  2 |  3 |  3
 13 |     12 |     17 |    1 |           -1 |      100 |               -1 |  3 |  3 |  4 |  3
  3 |     10 |     15 |   -1 |            1 |       -1 |              130 |  3 |  1 |  4 |  1
  2 |      6 |     10 |   -1 |            1 |       -1 |              100 |  2 |  1 |  3 |  1
(6 rows)
UPDATE edges SET cost = sign(cost), reverse_cost = sign(reverse_cost);
select * from edges where cost = -1 or reverse_cost = -1;
 id | source | target | cost | reverse_cost | capacity | reverse_capacity | x1 | y1 | x2 | y2 
----+--------+--------+------+--------------+----------+------------------+----+----+----+----
  5 |     10 |     11 |    1 |           -1 |      130 |               -1 |  3 |  1 |  3 |  2
 12 |      8 |     12 |    1 |           -1 |      100 |               -1 |  2 |  3 |  3 |  3
 11 |     11 |     12 |    1 |           -1 |      130 |               -1 |  3 |  2 |  3 |  3
 13 |     12 |     17 |    1 |           -1 |      100 |               -1 |  3 |  3 |  4 |  3
  3 |     10 |     15 |   -1 |            1 |       -1 |              130 |  3 |  1 |  4 |  1
  2 |      6 |     10 |   -1 |            1 |       -1 |              100 |  2 |  1 |  3 |  1
(6 rows)
```

**Comment by cvvergara** [comment]
> I ran a test with unequal directional costs and I receive a segmentation fault.

@iosefa can you post the simple test that you ran please

**Comment by cvvergara** [comment]
Knowing that the code creates a segmentation fault we can not merge.
This branch is on the main repo, so the branch is going to be deleted also.
